### PR TITLE
Add blank .js.rst file extension handler.

### PR DESCRIPTION
**Pull request by daithiocrualaoich**
33df9b57873

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "devDependencies": {
+    "strip-bom": "^3.0.0"
```
Comment (passy) [inline package.json]: wat

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+// potential for conflicts when a suffix is used by two modules simultaneously.
+// It also encourages the offloading of compilation to runtime instead of build
+// time.
+//
+// Strictly speaking it should not be used. However:
+//
+// - Only this module uses ``.js.rst`` suffixes.
+// - The Node module containing the ``require.extensions`` implementation is
+//   locked so Node is committed to keeping the current form indefinitely.
+// - Production use cases are encouraged to precompile ``.js.rst`` sources.
```
Comment (passy) [inline src/index.js:33]: I'm okay with this reasoning.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+// .. code-block:: javascript
+
+    const fs = require('fs')
+    const stripBom = require('strip-bom')
+
+    const loadFile = function(module, filename) {
+      var content = fs.readFileSync(filename, 'utf8')
+      content = stripBom(content)
+
+      console.log(content)
```
Comment (passy) [inline src/index.js]: So I reckon this is temporary?

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+// compilation and include the output into the running environment.
+//
+// .. code-block:: javascript
+
+    const fs = require('fs')
+    const stripBom = require('strip-bom')
+
+    const loadFile = function(module, filename) {
+      var content = fs.readFileSync(filename, 'utf8')
+      content = stripBom(content)
```
Comment (passy) [inline src/index.js]: Jesus, why is this necessary?

Also, is the reassignment necessary? Your project, but can't help myself from whinging about mutation.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        if (extensions[0] === '') {
+          extensions.shift()
+        }
+
+// Start with the longest possible extension and work towards the shortest.
+//
+// .. code-block:: javascript
+
+        while (extensions.shift()) {
+          var current = '.' + extensions.join('.')
```
Comment (passy) [inline src/index.js]: `let`?

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+//
+// This implementation follows the CoffeeScript directly.
+//
+// .. code-block:: javascript
+
+      const path = require('path')
+      const Module = require('module')
+
+      function findExtension(filename) {
+        var extensions = path.basename(filename).split('.')
```
Comment (passy) [inline src/index.js]: `let`?

**Review by passy — APPROVED** [review]
Your lack of semicolons may upset some people.

File: package.json (hunk 1 above)
Comment (daithiocrualaoich) [reply package.json]: I may have gotten a bit excited, sorry. The file content reading code I cut and pasted from StackOverflow was keen on UTF-8. I'll remove that.

File: src/index.js (hunk 2 above)
Comment (daithiocrualaoich) [reply src/index.js:33]: I'm working on a bin script to make the precompilation feasible today. This require hook business will be put in a separate `literacy/register` so it will have to be explicitly included. My intended use case is a Webpack loader anyway so the precompilation will happen there really. This is all following the CoffeeScript model.

Good doggies.

File: src/index.js (hunk 4 above)
Comment (daithiocrualaoich) [reply src/index.js]: Removed the `stripBom`. Wasn't watching the cut and paste and the encoding slipped in on the previous line.

Oh, but it's not real state mutation, Passy. :) Would I do that to you? This is the Rust trick where you reassign the variable name when you are finished with the previous value so the borrowchecker shuts up. We're done with the original contents of the file at this point and you can't have it again.

Anyways, it's gone.

File: src/index.js (hunk 3 above)
Comment (daithiocrualaoich) [reply src/index.js]: Yes, #2 extracts the JavaScript blocks from `content` and passed them to `Module._compile` instead of this.

File: src/index.js (hunk 6 above)
Comment (daithiocrualaoich) [reply src/index.js]: Definitely! Thanks for the teaching moment, I'm going to remember to do this in future.

**Review by daithiocrualaoich — COMMENTED** [review]
Thanks for the comments, really appreciated.

I've put in the semicolons and promise to find myself a linter.
